### PR TITLE
Authenticate Rancher admins as master

### PR DESCRIPTION
**Pull request by joshwget**
rancher/rancher#9468

### CODE REVIEW

File: authentication/rancher/authentication.go [hunk 1]
```diff
@@ -150,3 +156,36 @@ func (p *Provider) authDisabled() bool {
 
 	return setting.Value == "false"
 }
+
+func (p *Provider) isAdmin(token string) bool {
+	req, err := http.NewRequest("GET", p.url+"/accounts", nil)
```
Comment (alena1108) [inline authentication/rancher/authentication.go]: @joshwget any particular reason for not returning an error to a caller stack? here and on all lines below where error just gets translated to false

Comment (joshwget) [reply authentication/rancher/authentication.go]: Actually yeah, returning errors would be better. I did it this way when I thought accessing `/accounts` would give a 401.

Comment (joshwget) [reply authentication/rancher/authentication.go]: Updated

**Comment by alena1108** [comment]
LGTM

**Pull request merged**